### ISSUE TITLE
Exception: Cannot read property 'stat' of undefined

### ISSUE THREAD

**Issue by llaszkie** [webpack-contrib/cache-loader#43]
1. Latest version: **1.2.3**
1. Works fine if downgraded to 1.2.2 (via explicit dev-dependency)
1. The error occurs during **vue-service-cli build** for a project

Please check the declaration of module dependency. 

```
Module build failed (from ./node_modules/thread-loader/dist/cjs.js):
Thread Loader (Worker 10)
 Cannot read property 'stat' of undefined
     at toDepDetails (C:\agent\_work\2\s\node_modules\cache-loader\dist\index.js:65:14)
     at arrayIterator (C:\agent\_work\2\s\node_modules\neo-async\async.js:3780:9)
     at timesSync (C:\agent\_work\2\s\node_modules\neo-async\async.js:2292:7)
     at Object.mapLimit (C:\agent\_work\2\s\node_modules\neo-async\async.js:3775:5)
     at Array.<anonymous> (C:\agent\_work\2\s\node_modules\cache-loader\dist\index.js:88:18)
	 at arrayEachFunc (C:\agent\_work\2\s\node_modules\neo-async\async.js:2512:19)
     at Object.parallel (C:\agent\_work\2\s\node_modules\neo-async\async.js:6867:9)
     at Object.loader (C:\agent\_work\2\s\node_modules\cache-loader\dist\index.js:87:9)

```

**Comment by evilebottnawi** [comment]
@llaszkie can you create minimum reproducible test repo? Looks not all `fs` have `stat` method

**Comment by sodatea** [comment]
@evilebottnawi See https://github.com/vuejs/vue-cli/issues/2847
The missing of `this.fs` is caused by `thread-loader` if i'm not mistaken.

**Comment by evilebottnawi** [comment]
@sodatea strange, feel free to investigate and send a PR with fix, really very strange

**Comment by itmayziii** [comment]
Experienced this issue with the @Vue/cli when creating a new project and trying to build

**Comment by harishdurga** [comment]
yes i also have the same issue while trying to run build for the project

**Comment by Mitscherlich** [comment]
I think the `this.fs.stat` here should be `fs.stat`:

https://github.com/webpack-contrib/cache-loader/blob/91d2a1856748f0494971057efe114db35ddb1a28/src/index.js#L43-L47

I came across this problem when I build my app on travis CI. Since I didn't push yarn lock file, which reference to cache-loader `1.2.2`. And that's why my local test passed.

**Comment by evilebottnawi** [comment]
Please don't spam `i also have the same issue`, just use :+1: or better send a PR with fix

**Comment by evilebottnawi** [comment]
Somebody can create minimum reproducible test repo?

**Comment by gusvargas** [comment]
This will be resolved by by https://github.com/webpack-contrib/cache-loader/pull/45

But I wonder if this is actually a bug in the `thread-loader` instead, either in the code or in the documentation.

[this.fs](https://github.com/webpack-contrib/cache-loader/pull/45) is [clearly documented in the loader API](https://webpack.js.org/api/loaders/#this-fs). It seems like an odd precedent to set that every webpack loader needs to be thoroughly tested in the context of all other webpack loaders, especially when some webpack loaders essentially erase part of the API.

I think that the `thread-loader` should either make sure that `this.fs`is exposed for loaders running in a worker or document [in the usage](https://github.com/webpack-contrib/thread-loader#usage) that loaders being run in a worker cannot depend on `this.fs`.

**Comment by michael-ciniawsky** [comment]
Fixed by #45. Released in [`v1.2.4`](https://github.com/webpack-contrib/cache-loader/releases/tag/v1.2.4) 🎉
